### PR TITLE
Move Ctrl-F12 keybindings outside of libservo

### LINKED ISSUE #17647: Move Ctrl-F12 keybindings outside of libservo
This code should be removed from here: https://github.com/servo/servo/blob/95a85a301f1d014c45448a4d5d4f5a8497904af6/components/compositing/compositor.rs#L1316

Catching the keybinding should be done here: https://github.com/servo/servo/blob/95a85a301f1d014c45448a4d5d4f5a8497904af6/ports/glutin/window.rs#L1293

And new WindowEvent should be created here: https://github.com/servo/servo/blob/95a85a301f1d014c45448a4d5d4f5a8497904af6/components/compositing/windowing.rs#L34

And caught here: https://github.com/servo/servo/blob/95a85a301f1d014c45448a4d5d4f5a8497904af6/components/compositing/compositor.rs#L738 where we can actually toggle the profiler.

### PR DESCRIPTION
**Pull request by Tremoneck**
Creates a new WindowEvent called ToggleProfiller, when this Event is queued the profiler is toggled. 
When Crtl + F12 are pressed the Event is queued.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #17647 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17652)

### CODE REVIEW

File: components/compositing/windowing.rs [hunk 1]
```diff
@@ -76,6 +76,8 @@ pub enum WindowEvent {
     KeyEvent(Option<char>, Key, KeyState, KeyModifiers),
     /// Sent when Ctr+R/Apple+R is called to reload the current page.
     Reload,
+    //Toggles the profiler on and off
```
Comment (paulrouget) [inline components/compositing/windowing.rs]: Use 3 slashes + space: `//` -> `/// `

File: components/compositing/compositor.rs [hunk 2]
```diff
@@ -825,6 +825,11 @@ impl<Window: WindowMethods> IOCompositor<Window> {
                     warn!("Sending reload to constellation failed ({}).", e);
                 }
             }
+
+            WindowEvent::ToggleProfiler => {
```
Comment (paulrouget) [inline components/compositing/compositor.rs]: Can you rename this to ToggleWebRenderProfiler? Because their is also a regular profiler.

**Review by paulrouget — CHANGES_REQUESTED** [review]
Maybe use a better commit message: "Move WR profiler keybindings to /ports/"

File: ports/glutin/window.rs [hunk 3]
```diff
@@ -1290,6 +1290,9 @@ impl WindowMethods for Window {
                     self.event_queue.borrow_mut().push(WindowEvent::Quit);
                 }
             }
+            (CMD_OR_CONTROL, None, Key::F12) => {
```
Comment (paulrouget) [inline ports/glutin/window.rs]: Use just "CONTROL" here, not "CMD_OR_CONTROL".

**Comment by paulrouget** [comment]
Also:

I realize this is probably not necessary anymore: 

https://github.com/servo/servo/blob/2475039e40ec898e377c433836edafa5637d63ae/components/servo/lib.rs#L247 and https://github.com/servo/servo/blob/2475039e40ec898e377c433836edafa5637d63ae/components/compositing/compositor.rs#L1642 - remote these functions.

… and when you catch `WindowEvent::ToggleProfiler`, after calling `self.webrender.set_profiler_enabled`, you want to refresh WR, so add this: `self.webrender_api.generate_frame(None);`

**Comment by emilio** [comment]
@bors-servo delegate=paulrouget 

 * Someday I'll learn to write

File: components/compositing/windowing.rs (hunk 1 above)
Comment (Tremoneck) [inline components/compositing/windowing.rs]: Is there a use for just `//`, if not maybe this schould be added to `test-tidy`

Comment (paulrouget) [inline components/compositing/windowing.rs]: Yes. double slashes are for regular comments. triple are for documentation.

Comment (Tremoneck) [inline components/compositing/windowing.rs]: Ah ok, that makes sense

**Comment by Tremoneck** [comment]
I just run the failing tests `test-unit` myself, and the compiling of style_tests, which fails in travis, works for me. Is here a way to rerun the test?

**Comment by paulrouget** [comment]
@Tremoneck that's just a travis issue.

The retry will happen once you push again (or explicitly ask for a retry).

Your changes look good. Once you address https://github.com/servo/servo/pull/17652#issuecomment-314303849 it should be fine.

**Review by paulrouget — APPROVED** [review]
@bors-servo r+

This looks good to me. Thank you.

**Comment by paulrouget** [comment]
Thank you @Tremoneck

**Comment by Tremoneck** [comment]
I enjoyed it and will probably start doing more, when I have more time available

**Pull request merged**